survey_resource_booking/models/resource_booking.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 Tecnativa - Jairo Llopis
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.tests.common import Form
+from odoo.exceptions import UserError
+
+
+class ResourceBooking(models.Model):
+    _inherit = "resource.booking"
+
+    survey_user_input_id = fields.Many2one(
+        "survey.user_input",
+        string="Survey user input",
+        track_visibility="onchange",
+        help="User responses to survey, as specified in the resource booking type.",
+    )
+
+    def action_invite_survey(self):
+        """Invite requester to fill survey, if needed and possible."""
+        for one in self:
+            # Skip unconfirmed bookings, or finished user inputs
+            if one.state != "confirmed":
+                raise UserError(
+                    _(
+                        "Cannot invite to fill survey because booking not confirmed: %s"
+                    )
+                    % one.display_name
+                )
+            if one.survey_user_input_id.state == "done":
+                raise UserError(
+                    _(
+                        "This booking's requester was already invited to fill survey: %s"
+                    )
+                    % one.display_name
+                )
+            if not one.survey_user_input_id:
```

Comment:
Shouldn't the first check be after the second one? What if `survey_user_input_id` does not exist? 
```suggestion
            if not one.survey_user_input_id:
                # ....
            if one.survey_user_input_id.state == "done":
                raise UserError(
                    _(
                          "This booking's requester was already invited to fill survey: %s"
                      )
                      % one.display_name
                 )
```

Reply:
Well, it seems misleading but it's OK.

In Odoo, when you access a zero-recordset field, it will always be `False` (or an empty recordset, if the field is relational).